tests/upgrades/test_performance_tuning.py

```diff
@@ -0,0 +1,195 @@
+    "# Added by foreman-installer during upgrade, run the "
+    "installer with --upgrade-mongo-storage to upgrade to WiredTiger.",
+    "mongodb::server::storage_engine: 'mmapv1'"
+]
+
+WIREDTIGER_MONGODB = [
+    "# Do not remove",
+    "mongodb::server::storage_engine: 'wiredTiger'"
+]
+
+
+class ScenarioPerfornaceTuning(TestCase):
+    """The test class contains pre-upgrade and post-upgrade scenarios to test
+    Performance Tuning utility
+
+    Test Steps:
+
+    1. Before Satellite upgrade, we will apply the medium tune size.
+    2. Once the size get apply we will wait for post upgrade test case to check
+    whether the size is same or not.
+    3. In post upgrade we will check whether satellite installer was able to change
+    the size or not
+
+    """
+    @staticmethod
+    def _custom_hiera_file_creation(mongodb_type, tune_size):
+        """
+        This function we will use to create a custom-hiera.yaml file
+        as per users requirement"
+        :param int mongodb_type: Use to select  MMPV1 or WiredTiger data.
+        :param str tune_size: tune size would be medium, large, extra-large,
+        extra-extra-large.
+        """
+        with open('custom-hiera.yaml', 'w') as fd:
+            fd.write("\n".join(DEFAULT_CUSTOM_HIERA_DATA))
+            if mongodb_type:
+                fd.write("\n".join(MMPV1_MONGODB))
+            else:
+                fd.write("\n".join(WIREDTIGER_MONGODB))
+            if tune_size == "medium":
+                fd.write("\n".join(MEDIUM_TUNING_DATA))
+
+    @pre_upgrade
+    def test_pre_custom_hiera_file_validation(self):
+        """ In preupgrade scenario we will validate the default custom hiera file
+        parameter with expected custom_hiera file.
+        """
+        actual_custom_hiera = ssh.command("cat /etc/foreman-installer/custom-hiera.yaml").stdout
+        actual_custom_hiera = set([data.strip() for data in actual_custom_hiera])
+        cmd = 'grep "mongodb::server::storage_engine: \'wiredTiger\'" ' \
+              '/etc/foreman-installer/custom-hiera.yaml'
+        mongodb_type = ssh.command('{}'.format(cmd)).return_code
+        if mongodb_type:
+            expected_custom_hiera = set(DEFAULT_CUSTOM_HIERA_DATA + WIREDTIGER_MONGODB)
+        else:
+            expected_custom_hiera = set(DEFAULT_CUSTOM_HIERA_DATA + MMPV1_MONGODB)
+
+        assert expected_custom_hiera == actual_custom_hiera
+
+    @pre_upgrade
+    def test_pre_performance_tuning_apply(self):
+        """In preupgrade scenario we will apply the medium tuning size.
+
+        :id: preupgrade-83404326-20b7-11ea-a370-48f17f1fc2e1
+
+        :steps:
+            1. Run satellite-installer --disable-system-checks.
+
+        :expectedresults: Medium tuning parameter should be applied.
+
+         """
+        cmd = 'grep "mongodb::server::storage_engine: \'wiredTiger\'" ' \
+              '/etc/foreman-installer/custom-hiera.yaml'
+        mongodb_type = ssh.command('{}'.format(cmd)).return_code
```

Comment:
`cmd` is a string already, there is no reason to use it inside `format` to cast it to string.

Reply:
Fixed.